torch/distributed/tensor/_random.py

```diff
@@ -260,23 +255,48 @@ def _set_pre_op_offset(self, spec: DTensorSpec) -> None:
         """
         dtensor_shape = spec.shape
         mesh = spec.mesh
-        dim_map = spec.dim_map
+        # note: dim_map does not allow double sharding which is the FSDP(fully_shard)+TP
+        # case. Replace the custom logic with dim_map once we support it.
+        dim_map: List[Union[int, List[int]]] = [-1] * spec.ndim
```

Comment:
so dim_map will contain a mapping of 
"mesh dim" -> "tensor dims that are sharded over this mesh dim"
?

Reply:
the opposite. `dim_map` is "tensor dim" -> "mesh dims that the tensor dim is sharded over".